enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/EnterpriseSecurityModuleTest.java

```diff
@@ -19,83 +19,155 @@
+    {
+        // Given
+        nativeAuth( false, true );
+        ldapAuth( false, false );
+        pluginAuth( false, false );
+        authProviders( SecuritySettings.NATIVE_REALM_NAME );
 
         // Then
-        verify( mockLog, atLeastOnce() ).debug( anyString(),
-                contains( "Illegal configuration: No valid security realm is active." ), anyString() );
+        thrown.expect( IllegalArgumentException.class );
+        thrown.expectMessage( "Illegal configuration: All authentication providers are disabled." );
+
+        // When
+        new EnterpriseSecurityModule().newAuthManager( config, mockLogProvider, mock( SecurityLog.class), null, null );
+    }
+
+    @Test
+    public void shouldFailOnNoAuthorizationMechanism()
+    {
+        // Given
+        nativeAuth( true, false );
+        ldapAuth( false, false );
+        pluginAuth( false, false );
+        authProviders( SecuritySettings.NATIVE_REALM_NAME );
+
+        // Then
+        thrown.expect( IllegalArgumentException.class );
+        thrown.expectMessage( "Illegal configuration: All authorization providers are disabled." );
+
+        // When
+        new EnterpriseSecurityModule().newAuthManager( config, mockLogProvider, mock( SecurityLog.class), null, null );
     }
 
     @Test
     public void shouldFailOnIllegalAdvancedRealmConfiguration()
     {
         // Given
-        Config config = mock( Config.class );
-        LogProvider mockLogProvider = mock( LogProvider.class );
-        Log mockLog = mock( Log.class );
-        when( mockLogProvider.getLog( anyString() ) ).thenReturn( mockLog );
-        when( mockLog.isDebugEnabled() ).thenReturn( true );
-        when( config.get( SecuritySettings.native_authentication_enabled ) ).thenReturn( false );
-        when( config.get( SecuritySettings.native_authorization_enabled ) ).thenReturn( false );
-        when( config.get( SecuritySettings.ldap_authentication_enabled ) ).thenReturn( false );
-        when( config.get( SecuritySettings.ldap_authorization_enabled ) ).thenReturn( false );
-        when( config.get( SecuritySettings.plugin_authentication_enabled ) ).thenReturn( true );
-        when( config.get( SecuritySettings.plugin_authorization_enabled ) ).thenReturn( true );
-        when( config.get( SecuritySettings.auth_providers ) ).thenReturn(
-                Arrays.asList(
-                        SecuritySettings.NATIVE_REALM_NAME,
-                        SecuritySettings.LDAP_REALM_NAME )
-        );
+        nativeAuth( false, false );
+        ldapAuth( false, false );
+        pluginAuth( true, true );
+        authProviders( SecuritySettings.NATIVE_REALM_NAME, SecuritySettings.LDAP_REALM_NAME );
+
+        // Then
         thrown.expect( IllegalArgumentException.class );
+        thrown.expectMessage( "Illegal configuration: Native auth provider configured, " +
+                                "but both authentication and authorization are disabled." );
```

Comment:
I also favor easier debugging over the convenience of only changing one setting when re-configuring. I would assume changing config settings is not common in real deployment (only during testing and trying out by plugin authors), and the requirement to enable it in two places (order and boolean) is not so bad.